src/client.ts

```diff
@@ -0,0 +1,1832 @@
+import {
+	AddTagOptions,
+	BaseMongoObject,
+	ClassType,
+	EntityHistoric,
+	MongoClientConfiguration,
+	RemoveTagOptions,
+	StringMap,
+	UpdateManyAtOnceOptions,
+	UpdateManyQuery,
+} from './models';
+import { MongoClientUpdateManyOptions } from './models/update-many-options.models';
+import { UpdateManyToSameValueOptions } from './models/update-many-to-same-value-options.models';
+import { MongoReadStream } from './mongo-read-stream';
+import { MongoUtils } from './mongo-utils';
+import { TagManager } from './tag-manager';
+
+const defaultConfiguration: MongoClientConfiguration = {
+	keepHistoric: false,
+	historicPageSize: 100,
+};
+
+export class MongoClient<U extends BaseMongoObject, L extends BaseMongoObject> {
+	private readonly collection: Collection<U>;
+	private readonly collectionSourceForAggregation: Collection<U>;
+	private readonly logger: N9Log;
+	private readonly db: Db;
+	private readonly mongoClient: MongodbClient;
+	private readonly type: ClassType<U>;
+	private readonly typeList: ClassType<L>;
+	private readonly conf: MongoClientConfiguration;
+	private readonly lockFieldsManager: LockFieldsManager<U>;
+	private readonly indexManager: IndexManager;
+	private readonly historicManager: HistoricManager<U>;
+	private readonly tagManager: TagManager;
+
+	constructor(
+		collection: Collection<U> | string,
+		type: ClassType<U>,
+		typeList: ClassType<L>,
+		conf: MongoClientConfiguration = {},
+	) {
+		this.conf = _.merge({}, defaultConfiguration, conf);
+		if (this.conf.keepHistoric) {
+			this.conf.updateOnlyOnChange = { ...this.conf.updateOnlyOnChange };
+		}
+		if (this.conf.lockFields) {
+			this.lockFieldsManager = new LockFieldsManager(this.conf.lockFields);
+		}
+		this.logger = (global.log as N9Log).module('mongo-client');
+
+		this.db = global.db as Db;
+		if (!this.db) {
+			throw new N9Error('missing-db', 500);
+		}
+
+		this.mongoClient = global.dbClient as MongodbClient;
+		if (!this.mongoClient) {
+			throw new N9Error('missing-db-client', 500);
+		}
+
+		this.type = type;
+		this.typeList = typeList;
+
+		if (typeof collection === 'string') {
+			this.collection = this.db.collection(collection);
+		} else {
+			this.collection = collection;
+		}
+		this.indexManager = new IndexManager(this.collection);
+		this.historicManager = new HistoricManager(this.collection);
+		this.tagManager = new TagManager(this.collection);
+
+		if (this.conf.aggregationCollectionSource) {
+			this.collectionSourceForAggregation = this.db.collection(
+				this.conf.aggregationCollectionSource,
+			);
+		} else {
+			this.collectionSourceForAggregation = this.collection;
+		}
+	}
+
+	public async renameCollection(
+		newName: string,
+		dropTarget: boolean = false,
+		options?: { session?: ClientSession },
+	): Promise<MongoClient<U, L>> {
+		await this.collection.rename(newName, { ...options, dropTarget });
+		return new MongoClient<U, L>(newName, this.type, this.typeList, this.conf);
+	}
```

Comment:
```suggestion
	public async renameCollection(
		newName: string,
		options?: { dropTarget?: boolean; session?: ClientSession },
	): Promise<MongoClient<U, L>> {
		await this.collection.rename(newName, options);
		return new MongoClient<U, L>(newName, this.type, this.typeList, this.conf);
	}
```
A voir si on vraiment besoin de retourner un new MongoClient (on peut ajouter un parametre returnNewClient si nessesair)